azurerm/helpers/azure/hdinsight.go

```diff
@@ -45,6 +48,60 @@ func SchemaHDInsightTier() *schema.Schema {
 	}
 }
 
+func SchemaHDInsightSecurity() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"enable_enterprise_security_package": {
+					Type:     schema.TypeBool,
+					Optional: true,
+					Default:  false,
+					ForceNew: true,
+				},
+				"domain_username": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"cluster_users_group_dns": {
+					Type:     schema.TypeSet,
+					Optional: true,
+					Elem: &schema.Schema{
+						Type:         schema.TypeString,
+						ValidateFunc: validate.NoEmptyStrings,
+					},
+				},
+				"aadds_resource_id": {
+					Type:     schema.TypeString,
+					Computed: true,
```

Comment:
in my former idea, because there is only one aadds for a tenant, so we could get it through the rest api. so I added the dependency of `domainclient`.

I will follow your suggestion, let user inputs aadds, in this way, we don't need to add this dependency